File: frontend/src/interfaces/Telemedicine.ts
export interface IChatRoomPreview {
  id: number;
  participant_name: string;
  participant_picture_url: string;
  expired_at: string;
  last_chat: IChat;
}

export interface IChat {
  id: number;
  room_id: number;
  sender_account_id: number;
  message: string;
  attachment: IAttachment;
  prescription: IPrescription;
  created_at: string;
}

export interface IAttachment {
  format: string;
  url: string;
}

export interface IPrescriptionDrug {
  id: number;
  drug: { id: number; name: string; image: string };
  quantity: number;
  note: string;
}

export interface IPrescription {
  id: number;
  prescription_drugs: IPrescriptionDrug[];
}

export interface IChatRoom {
  id: number;
  doctor_account_id: number;
  user_account_id: number;
  doctor_certificate_url: string;
  expired_at: string;
  chats: IChat[];
}

File: backend/entity/telemedicine.go
package entity

import (
	"time"
)

type Chat struct {
	Id              int64
	RoomId          int64
	SenderAccountId int64
	Message         *string
	Attachment      Attachment
	Prescription    Prescription
	CreatedAt       *string
}

type Attachment struct {
	Url    *string
	Format *string
}

type ChatRoom struct {
	Id                   int64
	DoctorAccountId      int64
	UserAccountId        int64
	DoctorCertificateUrl string
	ExpiredAt            *time.Time
	ExpiredAtString      *string
	Chats                []Chat
}

type WsChatRoom struct {
	Id                   int64
	Hash                 string
	DoctorAccountId      int64
	UserAccountId        int64
	DoctorCertificateUrl string
	ExpiredAt            *int64
	Chats                []Chat
}

type ChatRoomPreview struct {
	Id                    int64
	ParticipantName       string
	ParticipantPictureUrl string
	ExpiredAt             *time.Time
	LastChat              Chat
}

type WsChatRoomPreview struct {
	Id                    int64
	Hash                  string
	ParticipantName       string
	ParticipantPictureUrl string
	ExpiredAt             *int64
	LastChat              Chat
}

type Participant struct {
	AccountId int64
	RoomId    int64
}
